library/java/io/envoyproxy/envoymobile/engine/EnvoyConfiguration.java

```diff
@@ -228,10 +232,19 @@ String resolveTemplate(final String configTemplate, final String platformFilterT
       h2RawDomainsAsString = sb.toString();
     }
 
-    String dnsResolverConfig = String.format(
-        "{\"@type\":\"type.googleapis.com/envoy.extensions.network.dns_resolver.cares.v3.CaresDnsResolverConfig\",\"resolvers\":%s,\"use_resolvers_as_fallback\": %s, \"filter_unroutable_families\": %s}",
-        dnsFallbackNameserversAsString, !dnsFallbackNameservers.isEmpty() ? "true" : "false",
-        dnsFilterUnroutableFamilies ? "true" : "false");
+    String dnsResolverName = "";
+    String dnsResolverConfig = "";
+    if (dnsUseSystemResolver) {
+      dnsResolverName = "envoy.network.dns_resolver.getaddrinfo";
+      dnsResolverConfig =
+          "{\"@type\":\"type.googleapis.com/envoy.extensions.network.dns_resolver.getaddrinfo.v3.GetAddrInfoDnsResolverConfig\"}";
+    } else {
+      dnsResolverName = "envoy.network.dns_resolver.cares";
+      dnsResolverConfig = String.format(
+          "{\"@type\":\"type.googleapis.com/envoy.extensions.network.dns_resolver.cares.v3.CaresDnsResolverConfig\",\"resolvers\":%s,\"use_resolvers_as_fallback\": %s, \"filter_unroutable_families\": %s}",
+          dnsFallbackNameserversAsString, !dnsFallbackNameservers.isEmpty() ? "true" : "false",
+          dnsFilterUnroutableFamilies ? "true" : "false");
```

Comment:
no action needed - just want to point out that for our new resolver `dnsFilterUnroutableFamilies` is effectively always equal to `true` as I pointed out in https://github.com/envoyproxy/envoy/pull/22080/files#r919203157. 

Just something for future us - we will need to clean up all of the DNS flags/options that we expose.